docs/source/reservations/existing-machines.rst

```diff
@@ -137,10 +139,49 @@ Example:
           identity_file: bob-key
           password: bob-password
 
-The ``~/.sky/ssh_node_pools.yaml`` file is stored on the server side, not client side:
+Apply ``~/.sky/sky_node_pools.yaml`` to the API server by the following steps for different setup:
+
+.. tab-set::
+
+   .. tab-item:: Local API server
+
+      If you did not start an API server instance or use a :ref:`local API server <sky-api-server-local>`, set ``~/.sky/ssh_node_pools.yaml`` on your local machine.
+   
+   .. tab-item:: Helm Deployment
+
+      If you use a :ref:`Helm Deployment <sky-api-server-helm-deploy-command>`, save the config to a ``ssh_node_pool.yaml`` file on your local machine and run:
+
+      .. code-block:: bash
+
+         # RELEASE_NAME and NAMESPACE are the same as the ones used in the helm deployment
+         helm upgrade --install $RELEASE_NAME skypilot/skypilot-nightly --devel \
+         --namespace $NAMESPACE \
+         --reuse-values \
+         --set-file apiService.sshNodePools=/your/path/to/ssh_node_pools.yaml
+      
+      If your `ssh_node_pools.yaml` requires SSH keys, you can create a secret that contains the keys and set the :ref:`apiService.sshKeySecret <helm-values-apiService-sshKeySecret>` to the secret name:
+
+      .. code-block:: bash
+
+         SECRET_NAME=apiserver-ssh-key
+         # The NAMESPACE should be consistent with the API server deployment
+         kubectl create secret generic $SECRET_NAME \
+            --namespace $NAMESPACE \
+            --from-file=id_rsa=/path/to/id_rsa \
+            --from-file=other_id_rsa=/path/to/other_id_rsa
+
+         helm upgrade --install $RELEASE_NAME skypilot/skypilot-nightly --devel \
+            --namespace $NAMESPACE \
+            --reuse-values \
+            --set apiService.sshKeySecret=$SECRET_NAME
```

Comment:
When the node pools specifies hosts from `~/.ssh/config`, it won't work even after the ssh key mount. Should we suggest the user to explicitly specify `ip`/`identity_file` in the `ssh_node_pools.yaml`?

Reply:
good catch, added a note